File: src/main/java/org/radarcns/consumer/realtime/action/appserver/SimpleTimeStrategy.java
package org.radarcns.consumer.realtime.action.appserver;

import java.time.Duration;
import java.time.Instant;
import java.time.temporal.ChronoUnit;

public class SimpleTimeStrategy implements ScheduleTimeStrategy {

  private final Duration delay;

  public SimpleTimeStrategy(long delay, ChronoUnit unit) {
    this.delay = Duration.of(delay, unit);
  }

  @Override
  public Instant getScheduledTime() {
    return Instant.now().plus(delay);
  }
}

File: src/main/java/org/radarcns/consumer/realtime/action/appserver/SimpleTimeStrategy.kt
package org.radarcns.consumer.realtime.action.appserver

import java.time.Duration
import java.time.Instant
import java.time.temporal.ChronoUnit

/** Schedules the time based on current time with an optional added delay.  */
class SimpleTimeStrategy(
        delay: Long,
        unit: ChronoUnit?,
        private val delayDuration: Duration = Duration.of(delay, unit)
) : ScheduleTimeStrategy {
    override val scheduledTime: Instant
        get() = Instant.now().plus(delayDuration)
}